lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -503,7 +503,7 @@ private function buildTableOptions(array $options)
             $options['collate'] = $options['charset'] . '_unicode_ci';
         }
 
-        $tableOptions[] = sprintf('COLLATE %s', $options['collate']);
+        $tableOptions[] = sprintf('COLLATE %s', $this->quoteSingleIdentifier($options['collate']));
```

Comment:
Can/should we call `$this->getColumnCollationDeclarationSQL()` here to reuse code?